tests/test_client/tests.py

```diff
@@ -85,6 +87,69 @@ def test_post(self):
         self.assertEqual(response.templates[0].name, 'POST Template')
         self.assertContains(response, 'Data received')
 
+    def test_json_post(self):
```

Comment:
Try refactoring the tests to using a loop and subTest and it doesn't look nice to duplicate so much code..

Reply:
As mentioned before I find doing this obfuscates what the tests are doing - especially when they are testing different things.

However if this is the standard in the lib I will do so.

Reply:
Hmmm to do this I'd have to pass in the test methods with some matching parameters which doesn't look elegant to me.

I've put the assertion code into a method and call that instead which means all these methods are now have 2 lines of code in them instead of repeating assertions.